### ISSUE TITLE
Open file from FUSE drive causes VS Code to freeze and put one CPU core at 100%

### ISSUE THREAD

**Issue by mbedded** [microsoft/vscode#172535]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.74.3
- OS Version:  Linux Mint 20.3 Cinnamon

Steps to Reproduce:

1. Mount your Cryptomator vault (using FUSE)
2. Open a Textfile from that Vault using VS Code
3. Application freezes and process has to be killed by hand

Additional hints:

- I already created a ticket here: https://github.com/microsoft/vscode/issues/149646. It was fixed someday so i closed the original ticket. A few weeks ago this started to happen again
- Other tools like vi, nano, xed or an IDE like Rider don't have such issues. They can open/read/write files from the vault without any issues

Main log:

```
2023-01-26 15:53:31.686 [error] [uncaught exception in main]: Error: write EPIPE
2023-01-26 15:53:31.686 [error] Error: write EPIPE
    at afterWriteDispatched (node:internal/stream_base_commons:160:15)
    at writeGeneric (node:internal/stream_base_commons:151:3)
    at Socket._writeGeneric (node:net:795:11)
    at Socket._write (node:net:807:8)
    at writeOrBuffer (node:internal/streams/writable:389:12)
    at _write (node:internal/streams/writable:330:10)
    at Writable.write (node:internal/streams/writable:334:10)
    at console.value (node:internal/console/constructor:286:16)
    at console.warn (node:internal/console/constructor:365:26)
    at d.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:2817)
    at v.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:4393)
    at Ct.u (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:53806)
    at I.a (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51349)
    at I.onUnexpectedError (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:249)
    at k (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:465)
    at process.<anonymous> (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51414)
    at process.emit (node:events:538:35)
    at process._fatalException (node:internal/process/execution:167:25)
2023-01-26 15:53:31.686 [error] [uncaught exception in main]: Error: write EPIPE
2023-01-26 15:53:31.686 [error] Error: write EPIPE
    at afterWriteDispatched (node:internal/stream_base_commons:160:15)
    at writeGeneric (node:internal/stream_base_commons:151:3)
    at Socket._writeGeneric (node:net:795:11)
    at Socket._write (node:net:807:8)
    at writeOrBuffer (node:internal/streams/writable:389:12)
    at _write (node:internal/streams/writable:330:10)
    at Writable.write (node:internal/streams/writable:334:10)
    at console.value (node:internal/console/constructor:286:16)
    at console.warn (node:internal/console/constructor:365:26)
    at d.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:2817)
    at v.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:4393)
    at Ct.u (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:53806)
    at I.a (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51349)
    at I.onUnexpectedError (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:249)
    at k (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:465)
    at process.<anonymous> (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51414)
    at process.emit (node:events:538:35)
    at process._fatalException (node:internal/process/execution:167:25)
2023-01-26 15:53:31.686 [error] [uncaught exception in main]: Error: write EPIPE
2023-01-26 15:53:31.686 [error] Error: write EPIPE
    at afterWriteDispatched (node:internal/stream_base_commons:160:15)
    at writeGeneric (node:internal/stream_base_commons:151:3)
    at Socket._writeGeneric (node:net:795:11)
    at Socket._write (node:net:807:8)
    at writeOrBuffer (node:internal/streams/writable:389:12)
    at _write (node:internal/streams/writable:330:10)
    at Writable.write (node:internal/streams/writable:334:10)
    at console.value (node:internal/console/constructor:286:16)
    at console.warn (node:internal/console/constructor:365:26)
    at d.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:2817)
    at v.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:4393)
    at Ct.u (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:53806)
    at I.a (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51349)
    at I.onUnexpectedError (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:249)
    at k (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:465)
    at process.<anonymous> (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51414)
    at process.emit (node:events:538:35)
    at process._fatalException (node:internal/process/execution:167:25)
2023-01-26 15:53:31.686 [error] [uncaught exception in main]: Error: write EPIPE
2023-01-26 15:53:31.686 [error] Error: write EPIPE
    at afterWriteDispatched (node:internal/stream_base_commons:160:15)
    at writeGeneric (node:internal/stream_base_commons:151:3)
    at Socket._writeGeneric (node:net:795:11)
    at Socket._write (node:net:807:8)
    at writeOrBuffer (node:internal/streams/writable:389:12)
    at _write (node:internal/streams/writable:330:10)
    at Writable.write (node:internal/streams/writable:334:10)
    at console.value (node:internal/console/constructor:286:16)
    at console.warn (node:internal/console/constructor:365:26)
    at d.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:2817)
    at v.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:4393)
    at Ct.u (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:53806)
    at I.a (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51349)
    at I.onUnexpectedError (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:249)
    at k (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:465)
    at process.<anonymous> (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51414)
    at process.emit (node:events:538:35)
    at process._fatalException (node:internal/process/execution:167:25)
2023-01-26 15:53:31.686 [error] [uncaught exception in main]: Error: write EPIPE
2023-01-26 15:53:31.686 [error] Error: write EPIPE
    at afterWriteDispatched (node:internal/stream_base_commons:160:15)
    at writeGeneric (node:internal/stream_base_commons:151:3)
    at Socket._writeGeneric (node:net:795:11)
    at Socket._write (node:net:807:8)
    at writeOrBuffer (node:internal/streams/writable:389:12)
    at _write (node:internal/streams/writable:330:10)
    at Writable.write (node:internal/streams/writable:334:10)
    at console.value (node:internal/console/constructor:286:16)
    at console.warn (node:internal/console/constructor:365:26)
    at d.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:2817)
    at v.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:4393)
    at Ct.u (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:53806)
    at I.a (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51349)
    at I.onUnexpectedError (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:249)
    at k (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:465)
    at process.<anonymous> (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51414)
    at process.emit (node:events:538:35)
    at process._fatalException (node:internal/process/execution:167:25)
2023-01-26 15:53:31.686 [error] [uncaught exception in main]: Error: write EPIPE
2023-01-26 15:53:31.686 [error] Error: write EPIPE
    at afterWriteDispatched (node:internal/stream_base_commons:160:15)
    at writeGeneric (node:internal/stream_base_commons:151:3)
    at Socket._writeGeneric (node:net:795:11)
    at Socket._write (node:net:807:8)
    at writeOrBuffer (node:internal/streams/writable:389:12)
    at _write (node:internal/streams/writable:330:10)
    at Writable.write (node:internal/streams/writable:334:10)
    at console.value (node:internal/console/constructor:286:16)
    at console.warn (node:internal/console/constructor:365:26)
    at d.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:2817)
    at v.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:4393)
    at Ct.u (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:53806)
    at I.a (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51349)
    at I.onUnexpectedError (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:249)
    at k (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:465)
    at process.<anonymous> (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51414)
    at process.emit (node:events:538:35)
    at process._fatalException (node:internal/process/execution:167:25)
2023-01-26 15:53:31.686 [error] [uncaught exception in main]: Error: write EPIPE
2023-01-26 15:53:31.686 [error] Error: write EPIPE
    at afterWriteDispatched (node:internal/stream_base_commons:160:15)
    at writeGeneric (node:internal/stream_base_commons:151:3)
    at Socket._writeGeneric (node:net:795:11)
    at Socket._write (node:net:807:8)
    at writeOrBuffer (node:internal/streams/writable:389:12)
    at _write (node:internal/streams/writable:330:10)
    at Writable.write (node:internal/streams/writable:334:10)
    at console.value (node:internal/console/constructor:286:16)
    at console.warn (node:internal/console/constructor:365:26)
    at d.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:2817)
    at v.error (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:39:4393)
    at Ct.u (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:53806)
    at I.a (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51349)
    at I.onUnexpectedError (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:249)
    at k (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:7:465)
    at process.<anonymous> (/usr/share/codium/resources/app/out/vs/code/electron-main/main.js:82:51414)
    at process.emit (node:events:538:35)
    at process._fatalException (node:internal/process/execution:167:25)

```

**Comment by deepak1556** [comment]
> Mount your Cryptomator vault (using FUSE)

I am not familiar with cryptomator, can you provide a minimal example that does not require installing third-party software. Thanks!

**Comment by mbedded** [comment]
Hello deepak,

unfortunately I have no sample yet to mount a fuse drive using CLI only. Maybe I can find one but I am no Linux expert. In Cryptomator I can choose Between FUSE and WebDAV to mount my directory.

Back then (as I wrote the original ticket) I did the following to identify where the issue comes from:

1. Newest Version of VS Code (disable all extensions), downgrade Cryptomator to an older version (I even put my computer back into an older version by using a Backup-Image to ensure the underlying libraries are older, too) but this didn't resolve the issue
2. Newest Version of Cryptomator and system-libraries, downgrade VS Code (disable all extensions) and then I saw that an older Version works like before.

As a side note: I use the combination FUSE on Windows using Dokan Library) and VS Code without any issues. I am not sure what has changed within VS Code on Linux.
I try my best to find a CLI-Example to mount a drive and reproduce the behavior. Using the mounted Cryptomator-Vault via FUSE is currently my only knowledge how to reproduce that. I understand your point of view to reproduce the issue without 3rd party software
